### PR TITLE
My location auto center

### PR DESCRIPTION
**Pull request by nd-novorender**
https://trello.com/c/bQ3F3Pb5/807-my-location-center-camera-on-marker

- Change marker for my location to the same as google maps have
- New "Center" option to automatically center on the marker
- Only show marker when "Marker" is enabled
- Support flyTime for ortho camera
- Chnage location min update rate 5000->500ms

### CODE REVIEW

File: src/features/myLocation/myLocationMarker.tsx [hunk 1]
```diff
@@ -38,14 +42,11 @@ export const MyLocationMarker = forwardRef<{ update: () => void }>(function MyLo
...
-            {location && status !== LocationStatus.Error && (
-                <path
-                    ref={setContainer}
-                    id="myLocationPoint"
-                    name="myLocationPoint"
-                    fill={theme.palette.primary.main}
-                    d="M12 2C8.14 2 5 5.14 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.86-3.14-7-7-7zm0 2c1.1 0 2 .9 2 2 0 1.11-.9 2-2 2s-2-.89-2-2c0-1.1.9-2 2-2zm0 10c-1.67 0-3.14-.85-4-2.15.02-1.32 2.67-2.05 4-2.05s3.98.73 4 2.05c-.86 1.3-2.33 2.15-4 2.15z"
-                />
+            {showMarker && location && status !== LocationStatus.Error && (
+                <g ref={setContainer} id="myLocationPoint" name="myLocationPoint">
```
Comment (barabadzhi) [inline src/features/myLocation/myLocationMarker.tsx:46]: I'm not keen on `myLocationPoint` as a name/id 🤔

File: src/features/myLocation/myLocationSlice.ts [hunk 2]
```diff
@@ -32,6 +33,9 @@ export const myLocationSlice = createSlice({
         toggleShowMarker: (state, action: PayloadAction<boolean | undefined>) => {
             state.showMarker = action.payload ?? !state.showMarker;
         },
+        toggleAutocenter: (state, action: PayloadAction<boolean>) => {
```
Comment (barabadzhi) [inline src/features/myLocation/myLocationSlice.ts:36]: Should this one work as toggle?Like `toggleShowMarker` above, `!!` the toggle if the value is not passed, so we have uniform behavior.

File: src/features/myLocation/myLocationSlice.ts [hunk 3]
```diff
@@ -43,6 +47,7 @@ export const myLocationSlice = createSlice({
 
 export const selectCurrentLocation = (state: RootState) => state.myLocation.currentLocation;
 export const selectShowLocationMarker = (state: RootState) => state.myLocation.showMarker;
+export const selectMyLocationAutocenter = (state: RootState) => state.myLocation.autocenter;
```
Comment (barabadzhi) [inline src/features/myLocation/myLocationSlice.ts:50]: Again, not sure if `myLocation` is the best name.

File: src/features/myLocation/myLocationSlice.ts (hunk 2 above)
Comment (nd-novorender) [reply src/features/myLocation/myLocationSlice.ts:36]: Good point, will update, but with `!`, not `!!`

File: src/features/myLocation/myLocationSlice.ts (hunk 3 above)
Comment (nd-novorender) [reply src/features/myLocation/myLocationSlice.ts:50]: You mean in `selectMyLocationAutocenter` or in `state.myLocation.autocenter`? What would you suggest?

Comment (barabadzhi) [reply src/features/myLocation/myLocationSlice.ts:50]: In both actually, I think `myLocation` is not a great name. I understand it was like that before, so feel free to leave it if you will.

**Review by barabadzhi — APPROVED** [review]

**Pull request merged**